### ISSUE TITLE
mechanical backlash in Y-axis

### ISSUE THREAD

**Issue by hanebefl** [fab-machines/Fabulaser-Mini#1]
## issue
The Fabulaser machines I've seen all have a small issue with what I (until now) assumed is backlash in the Y-axis. This is well visible with marking or cutting small circles: There are small "edges" at the top right and bottom left (when cutting clockwise).
![backlash_circle](https://user-images.githubusercontent.com/35426311/180656065-9d4757d6-63b1-4c28-99c2-05053dbfcd7b.png)

I've managed to pinpoint the location of the backlash to the couplers, connecting the Y-motor to the Y-shafts. 

## setup
To verify this location, I've attached two levers to a coupler (one at the motor side, one at the shaft side) and placed a dial indicator on them. 
![setup](https://user-images.githubusercontent.com/35426311/180653079-cc585ae0-8fbf-422c-bb02-2401abb0909e.jpg)

Next I've created a gcode file which moves the Y-axis back and forth ten times, first one step, then two steps, then three and so on. To not lose count of the number of steps, the X-axis moves for 5mm between each cycle. The gcode file and python script generating the gcode, are attached here:
[backlash-gcode-and-py.zip](https://github.com/fab-machines/Fabulaser-Mini/files/9176003/backlash-gcode-and-py.zip)

I've defined the backlash as first measurement that actually moves the needle of the dial indicator back and forth. Some movement in only one direction can be observed first, but this is just part of the backlash itself.

## findings
With this setup, the dial indicator shows a back-and-forth movement at the motor side at 4 steps (or 0.045 mm) travel. At the shaft side the back-and-forth movement is only detected at 15 or 16 steps (0.169 mm). This confirms a backlash of 0.124 millimeters in the couplers.

Here you can see two animations, with the number of steps moved (sorry for the shaky camera):
![motor_side](https://user-images.githubusercontent.com/35426311/180655930-0367b697-9602-45e6-9e1a-f330958f2704.gif)
![shaft_side](https://user-images.githubusercontent.com/35426311/180655934-f5bd11dd-103d-474f-a3eb-a97e80c13b01.gif)
In the second animation you can actually see movement of the left "motor" lever, while the right "shaft" lever is standing still.

## next steps
There seems to be a lot of resistance in the movement of the Y-axis. One area I've detected is the 'Y-pulley front'. I've got an idea for a different design, and will report my findings.

Maybe there's a different type of coupler, that less backlash at the same torque?

**Comment by satshas** [comment]
First of all **many thanks** for the time you took to test this out and also to verify it with the dial indicators.

Seems definitively as you argue, with the couplers flexing a bit while the Y axis is moving (assuming the guides+bearings are lubricated and the rest of the assembly is well done). The Y-Pulley front design is also worth a change to improve smoothness as it is hold only on one side. 

We're currently making the large format Laser and we took many of these parameters into account, even decoupling the Y motor from the Y shaft/belts. We'll probably share them in the next release of Fabulaser, if they prove to be good on the large design. In the meantime if your findings are good we can add them to the repository.

In the design of the CNC Milling machine we started using these couplers, maybe worth to try them also on Fabulaser?

![coupler](https://user-images.githubusercontent.com/6417889/180657894-a62a2bbe-ea83-496c-89a8-670cf939587b.jpg)
